asf_search/ASFProduct.py

```diff
@@ -59,6 +59,10 @@ def stack(
         """
         from .search.baseline_search import stack_from_product
 
+        if opts is None:
+            opts = ASFSearchOptions()
+            opts.session = self.session
```

Comment:
I think you can do this in a single line as `ASFSearchOptions(session=self.session)`